bindings/sundials4py/idas/idas_generated.hpp

```diff
@@ -0,0 +1,1881 @@
+                                                              ypS0_1d);
+  },
+  nb::arg("ida_mem"), nb::arg("yyS0_1d"), nb::arg("ypS0_1d"),
+  "Initial condition calculation function");
+
+m.def("IDASetSensDQMethod", IDASetSensDQMethod, nb::arg("ida_mem"),
+      nb::arg("DQtype"), nb::arg("DQrhomax"));
+
+m.def("IDASetSensErrCon", IDASetSensErrCon, nb::arg("ida_mem"),
+      nb::arg("errconS"));
+
+m.def("IDASetSensMaxNonlinIters", IDASetSensMaxNonlinIters, nb::arg("ida_mem"),
+      nb::arg("maxcorS"));
+
+m.def(
+  "IDASetSensParams",
+  [](void* ida_mem, sundials4py::Array1d p_1d, sundials4py::Array1d pbar_1d,
+     std::vector<int> plist_1d) -> int
+  {
+    auto IDASetSensParams_adapt_arr_ptr_to_std_vector =
+      [](void* ida_mem, sundials4py::Array1d p_1d, sundials4py::Array1d pbar_1d,
+         std::vector<int> plist_1d) -> int
+    {
+      sunrealtype* p_1d_ptr    = reinterpret_cast<sunrealtype*>(p_1d.data());
+      sunrealtype* pbar_1d_ptr = reinterpret_cast<sunrealtype*>(pbar_1d.data());
+      int* plist_1d_ptr =
+        reinterpret_cast<int*>(plist_1d.empty() ? nullptr : plist_1d.data());
+
+      auto lambda_result = IDASetSensParams(ida_mem, p_1d_ptr, pbar_1d_ptr,
+                                            plist_1d_ptr);
+      return lambda_result;
+    };
+
+    return IDASetSensParams_adapt_arr_ptr_to_std_vector(ida_mem, p_1d, pbar_1d,
+                                                        plist_1d);
+  },
+  nb::arg("ida_mem"), nb::arg("p_1d"), nb::arg("pbar_1d"), nb::arg("plist_1d"));
+
+m.def("IDASetNonlinearSolverSensSim", IDASetNonlinearSolverSensSim,
+      nb::arg("ida_mem"), nb::arg("NLS"));
+
+m.def("IDASetNonlinearSolverSensStg", IDASetNonlinearSolverSensStg,
+      nb::arg("ida_mem"), nb::arg("NLS"));
+
+m.def("IDASensToggleOff", IDASensToggleOff, nb::arg("ida_mem"),
+      "Enable/disable sensitivities");
+
+m.def(
+  "IDAGetSens",
+  [](void* ida_mem, std::vector<N_Vector> yySout_1d) -> std::tuple<int, sunrealtype>
+  {
+    auto IDAGetSens_adapt_arr_ptr_to_std_vector =
+      [](void* ida_mem, sunrealtype* tret, std::vector<N_Vector> yySout_1d) -> int
+    {
+      N_Vector* yySout_1d_ptr = reinterpret_cast<N_Vector*>(
+        yySout_1d.empty() ? nullptr : yySout_1d.data());
+
+      auto lambda_result = IDAGetSens(ida_mem, tret, yySout_1d_ptr);
+      return lambda_result;
+    };
+    auto IDAGetSens_adapt_modifiable_immutable_to_return =
+      [&IDAGetSens_adapt_arr_ptr_to_std_vector](void* ida_mem,
+                                                std::vector<N_Vector> yySout_1d)
+      -> std::tuple<int, sunrealtype>
+    {
+      sunrealtype tret_adapt_modifiable;
+
+      int r = IDAGetSens_adapt_arr_ptr_to_std_vector(ida_mem,
+                                                     &tret_adapt_modifiable,
+                                                     yySout_1d);
+      return std::make_tuple(r, tret_adapt_modifiable);
+    };
```

Comment:
Nothing wrong here. Just think it's interesting that it writes these a two separate functions rather than combining them.